### ISSUE TITLE
Can't dig +trace

### ISSUE THREAD

**Issue by adilias3** [pi-hole/FTL#383]
**In raising this issue, I confirm the following (please check boxes, eg [X]) Failure to fill the template will close your issue:**

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*
- [X] The issue I am reporting isn't a duplicate 

**How familiar are you with the codebase?:**

1

---

**[BUG] Expected Behaviour:**

```
dig +trace www.example.com

; <<>> DiG 9.10.6 <<>> +trace www.example.com
;; global options: +cmd
.			518400	IN	NS	a.root-servers.net.
.			518400	IN	NS	b.root-servers.net.
.			518400	IN	NS	c.root-servers.net.
.			518400	IN	NS	d.root-servers.net.
.			518400	IN	NS	e.root-servers.net.
```

dig +trace performs a query for root zone NS records, it should start by returning a list of root servers. As above.

**[BUG | ISSUE] Actual Behaviour:**

```
dig +trace www.example.com

; <<>> DiG 9.10.6 <<>> +trace www.example.com
;; global options: +cmd
;; Received 17 bytes from 172.28.28.25#53(172.28.28.25) in 12 ms
```
With packet capture, pihole-FTL returns 'Flags: 0x80a2 Standard query response, Server failure', and dig +trace fails.

**[BUG | ISSUE] Steps to reproduce:**

This is occurring on 2 different pihole installs, both pihole-FTL v4.0

If I stop the pihole-FTL service and start dnsmasq service then correct behaviour returns.

```
systemctl stop pihole-FTL && systemctl start dnsmasq
```
dig +trace is now working correctly.

```
systemctl stop dnsmasq && systemctl start pihole-FTL
```
dig +trace is now failing again as above example.

OS:

I have tested this on 2 different machines on up to date Ubuntu 18 installs.

pihole was installed via install script : curl -sSL https://install.pi-hole.net | bash

I updated pihole by running pihole -up on both servers and the issue persists.

**Comment by DL6ER** [comment]
Which version of `dnsmasq` did you use for testing? We do not interfere with the `NS` query type handling in the DNS resolver, so it may be an issue that was introduced sometime between the `dnsmasq` version you're using and the one embedded in `pihole-FTL` (incorporating version 2.79).

**Comment by adilias3** [comment]
```
# for dnsmasq
systemctl stop pihole-FTL && systemctl start dnsmasq
dig chaos txt version.bind +short
"dnsmasq-2.76"

#for pihole-FTL
systemctl stop dnsmasq && systemctl start pihole-FTL
dig chaos txt version.bind +short
"dnsmasq-pi-hole-2.79"
```

**Comment by DL6ER** [comment]
With `dnsmasq` v2.79 and v2.78, I see exactly the same behavior as with `pihole-FTL`:
```
; <<>> DiG 9.10.3-P4-Ubuntu <<>> +trace www.example.com @pi.hole
;; global options: +cmd
;; Received 17 bytes from 192.168.2.11#53(pi.hole) in 76 ms
```
With `dnsmasq` v2.77 and v2.76, I see:
```
; <<>> DiG 9.10.3-P4-Ubuntu <<>> +trace www.example.com @pi.hole
;; global options: +cmd
.			42569	IN	NS	l.root-servers.net.
.			42569	IN	NS	k.root-servers.net.
.			42569	IN	NS	e.root-servers.net.
.			42569	IN	NS	h.root-servers.net.
.			42569	IN	NS	j.root-servers.net.
.			42569	IN	NS	i.root-servers.net.
.			42569	IN	NS	g.root-servers.net.
.			42569	IN	NS	a.root-servers.net.
.			42569	IN	NS	b.root-servers.net.
.			42569	IN	NS	m.root-servers.net.
.			42569	IN	NS	c.root-servers.net.
.			42569	IN	NS	f.root-servers.net.
.			42569	IN	NS	d.root-servers.net.
;; Received 241 bytes from 192.168.2.11#53(pi.hole) in 115 ms
```

**TL;DR** The bug was introduced in `dnsmasq` v2.78. I will contact Simon Kelley, the author of `dnsmasq` about this and will report back here.

**Comment by DL6ER** [comment]
@adilias3 This has been confirmed as an original `dnsmasq` v2.79+ bug. We will incorporate the fix into `pihole-FTL` v4.1. Thanks for reporting this!

**Comment by DL6ER** [comment]
This issue will be fixed in FTL v4.2.
